### PR TITLE
[Word] (events) Add article

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**

### CODE REVIEW

File: docs/word/word-add-ins-events.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ms.localizationpriority: medium
+---
+
+# Work with events using the Word JavaScript API
+
+This article describes important concepts related to working with events in Word and provides code samples that show how to register event handlers, handle events, and remove event handlers using the Word JavaScript API.
+
+## Events in Word
+
+Each time certain types of changes occur in a Word document, an event notification fires. By using the Word JavaScript API, you can register event handlers that allow your add-in to automatically run a designated function when a specific event occurs. The following events are currently supported.
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:14]: I know the pluralization was correct, but I think simplifying the sentences a bit would help speed readers.

```suggestion
When certain changes occur in a Word document, event notifications fire. The Word JavaScript APIs let you register event handlers that allow your add-in to automatically run designated functions when those changes occur. The following events are currently supported.
```

File: docs/word/word-add-ins-events.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+| Event | Description | Supported objects |
+|:---------------|:-------------|:-----------|
+| `onAnnotationClicked` | Occurs when the user clicks an annotation (or selects it using **Alt+Down**).<br><br>Event data object:<br>[AnnotationClickedEventArgs](/javascript/api/word/word.annotationclickedeventargs) | [**Document**](/javascript/api/word/word.document#word-word-document-onannotationclicked-member) |
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:18]: Is the keyboard shortcut important here? If so, please ignore this.
```suggestion
| `onAnnotationClicked` | Occurs when the user selects an annotation.<br><br>Event data object:<br>[AnnotationClickedEventArgs](/javascript/api/word/word.annotationclickedeventargs) | [**Document**](/javascript/api/word/word.document#word-word-document-onannotationclicked-member) |
```

File: docs/word/word-add-ins-events.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+- VBA add-in (macro) code that changes the document.
+
+Any change that complies with default behavior of Word will trigger the corresponding events in a document.
+
+### Lifecycle of an event handler
+
+An event handler is created when an add-in registers the event handler. It's destroyed when the add-in deregisters the event handler or when the add-in is refreshed, reloaded, or closed. Event handlers don't persist as part of the Word file, or across sessions with Word on the web.
+
+> [!CAUTION]
+> When an object to which events are registered is deleted (e.g., a content control with an `onDataChanged` event registered), the event handler no longer triggers but remains in memory until the add-in or Word session refreshes or closes.
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:61]: Is this a cautionary statement because of memory issues?

File: docs/word/word-add-ins-events.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+### Events and coauthoring
+
+With coauthoring, multiple people can work together and edit the same Word document simultaneously. For events that can be triggered by a coauthor, such as `onParagraphChanged`, the corresponding **Event** object will contain a **source** property that indicates whether the event was triggered locally by the current user (`event.source = Local`) or was triggered by the remote coauthor (`event.source = Remote`).
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:65]: Should "triggered by coauthoring" be a table column entry?

Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:65]:
```suggestion
With coauthoring, multiple people can work together and edit the same Word document simultaneously. For events that can be triggered by a coauthor, such as `onParagraphChanged`, the corresponding **Event** object will contain a **source** property that indicates whether the event was triggered locally by the current user (`event.source == Local`) or was triggered by the remote coauthor (`event.source == Remote`).
```

File: docs/word/word-add-ins-events.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+## Register an event handler
+
+The following code sample registers an event handler for the `onParagraphChanged` event in the document. The code specifies that when content changes in the document, the `handleChange` function should run.
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:69]: Let's avoid ambiguity in when the event handler fires.
```suggestion
The following code sample registers an event handler for the `onParagraphChanged` event in the document. The code specifies that when content changes in the document, the `handleChange` function runs.
```

File: docs/word/word-add-ins-events.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+```js
+await Word.run(async (context) => {
+    eventContext = context.document.onParagraphChanged.add(handleChange);
+
+    await context.sync();
+    console.log("Event handler successfully registered for onParagraphChanged event in the document.");
+}).catch(errorHandlerFunction);
+```
+
+## Handle an event
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:80]: Consider merging this with the previous section. I'm not sure there's a benefit to splitting the code here.

File: docs/word/word-add-ins-events.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+  });
+}
+```
+
+## Use .track()
+
+Certain event types also require you to call `track()` on the object you're adding the event to.
+
+- Content control events
+  - onDataChanged
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:133]: Should these event names be in code ticks?

File: docs/word/word-add-ins-events.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    
+    eventContext = null;
+    console.log("Removed event handler that was tracking content changes in paragraphs.");
+  });
+}
+```
+
+## Use .track()
+
+Certain event types also require you to call `track()` on the object you're adding the event to.
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:130]: Consider avoiding ending the sentence in a preposition.
```suggestion
Certain event types also require you to call `track()` on the object to which the event is added.
```

**Review by AlexJerabek — COMMENTED** [review]
Looks good. Left a few comments for consideration while the PR is a draft.

File: docs/word/word-add-ins-events.md (hunk 3 above)
Comment (gergzk) [reply docs/word/word-add-ins-events.md:61]: What should a reader do with this information? I expect this is pretty esoteric/non-actionable for them? My inclination would be to skip this.

File: docs/word/word-add-ins-events.md (hunk 4 above)
Comment (gergzk) [reply docs/word/word-add-ins-events.md:65]: Have you definitely validated this works? My recollection from Excel is that coauth changes do not trigger events. Your add on line 50 does not include coauth changes - should it?

File: docs/word/word-add-ins-events.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+async function handleChange(event: Word.ParagraphChangedEventArgs) {
+  await Word.run(async (context) => {
+    await context.sync();
+    console.log(`${event.type} event was detected.`);
+  });
+}
+
+async function deregisterEventHandler() {
+  await Word.run(eventContext.context, async (context) => {
```
Comment (gergzk) [inline docs/word/word-add-ins-events.md:118]: Consider explicitly calling out that the add-in needs to use the same context object on which the handler was registered. This surprises a lot of people

File: docs/word/word-add-ins-events.md (hunk 8 above)
Comment (ElizabethSamuel-MSFT) [reply docs/word/word-add-ins-events.md:130]: Per [guidance](https://learn.microsoft.com/en-us/style-guide/grammar/prepositions#placement-in-the-sentence), we can end with a preposition. I might wait to see what Linda Lu says for this.

File: docs/word/word-add-ins-events.md (hunk 9 above)
Comment (ElizabethSamuel-MSFT) [reply docs/word/word-add-ins-events.md:129]: Added comment accordingly

File: docs/word/word-add-ins-events.md [hunk 10]
```diff
@@ -11,37 +11,37 @@ This article describes important concepts related to working with events in Word
...
-| `onParagraphDeleted` | Occurs when the user deletes paragraphs.<br><br>Event data object:<br>[ParagraphDeletedEventArgs](/javascript/api/word/word.paragraphdeletedeventargs) | [**Document**](/javascript/api/word/word.document#word-word-document-onparagraphdeleted-member) |
-| `onSelectionChanged` | Occurs when selection within the content control is changed.<br><br>Event data object:<br>[ContentControlSelectionChangedEventArgs](/javascript/api/word/word.contentcontrolselectionchangedeventargs) | [**ContentControl**](/javascript/api/word/word.contentcontrol#word-word-contentcontrol-onselectionchanged-member) |
+When certain changes occur in a Word document, event notifications fire. The Word JavaScript APIs let you register event handlers that allow your add-in to automatically run designated functions when those changes occur. The following events are currently supported.
+
+| Event | Description | Supported objects | Supports coauthoring? |
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:16]: Is "Supports coauthoring?" the right framing?  That seems to switch the direction of what triggers the event. Consider something like "Triggered during coauthoring?" to indicate that changes made by other users will trigger events in your session.
```suggestion
| Event | Description | Supported objects | Triggered during coauthoring? |
```

File: docs/word/word-add-ins-events.md [hunk 11]
```diff
@@ -11,37 +11,37 @@ This article describes important concepts related to working with events in Word
...
-| `onCommentSelected` | Occurs when a comment is selected.<br><br>Event data object:<br>[CommentEventArgs](/javascript/api/word/word.commenteventargs) | <ul><li>[**Body**](/javascript/api/word/word.body?view=word-js-preview&preserve-view=true#word-word-body-oncommentselected-member)</li><li>[**ContentControl**](/javascript/api/word/word.contentcontrol?view=word-js-preview&preserve-view=true#word-word-contentcontrol-oncommentselected-member)</li><li>[**Paragraph**](/javascript/api/word/word.paragraph#word-word-paragraph-oncommentselected-member)</li><li>[**Range**](/javascript/api/word/word.range#word-word-range-oncommentselected-member)</li></ul> |
+| Event | Description | Supported objects | Supports coauthoring? |
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:38]:
```suggestion
| Event | Description | Supported objects | Triggered during coauthoring? |
```

File: docs/word/word-add-ins-events.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+    }
+
+    await context.sync();
+
+    console.log("Added event handlers for when content controls are deleted.");
+  }
+});
+```
+
+The following code sample shows how to register comment event handlers on the document's body object and include a `body.track();` statement.
```
Comment (AlexJerabek) [inline docs/word/word-add-ins-events.md:184]: Nit: Personal preference on excluding semicolons when talking about code in text.
```suggestion
The following code sample shows how to register comment event handlers on the document's body object and include a `body.track()` statement.
```

**Review by AlexJerabek — APPROVED** [review]
Looks good! A couple comments to consider, but nothing blocking from me.

**Pull request merged**